File: resolve_conflicts.js
const fs = require('fs');

const filePath = 'd:/djtrip/dashbord/src/sections/Page1/Components/LieuDetails.jsx';
let content = fs.readFileSync(filePath, 'utf8');

// Try multiple regex patterns to handle different line endings
const patterns = [
  /<<<<<<< HEAD\n([\s\S]*?)\n=======\n[\s\S]*?\n>>>>>>> backend\/djtripx2/g,
  /<<<<<<< HEAD\r\n([\s\S]*?)\r\n=======\r\n[\s\S]*?\r\n>>>>>>> backend\/djtripx2/g
];

let conflictCount = 0;

// Try each pattern
for (const pattern of patterns) {
  const matches = content.match(pattern);
  if (matches && matches.length > 0) {
    conflictCount = matches.length;
    content = content.replace(pattern, '$1');
    console.log(`✓ Pattern matched with ${conflictCount} conflicts`);
    break;
  }
}

if (conflictCount === 0) {
  // Fallback: try a more permissive pattern
  const fallbackPattern = /<<<<<<< HEAD([\s\S]*?)=======([\s\S]*?)>>>>>>> backend\/djtripx2/g;
  const fallbackMatches = content.match(fallbackPattern);
  if (fallbackMatches && fallbackMatches.length > 0) {
    conflictCount = fallbackMatches.length;
    content = content.replace(fallbackPattern, function(match, head, backend) {
      return head;
    });
    console.log(`✓ Fallback pattern matched with ${conflictCount} conflicts`);
  }
}

// Write back to file
fs.writeFileSync(filePath, content, 'utf8');

console.log('✓ File: LieuDetails.jsx');
console.log('✓ Conflicts resolved: ' + conflictCount);
console.log('✓ Status: Success - File updated with HEAD version kept for all conflicts');

File: dashbord/resolve_conflicts.py
#!/usr/bin/env python3
import re
import sys
from pathlib import Path

def resolve_conflicts_keep_head(file_path):
    """Resolve merge conflicts by keeping HEAD version"""
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()

    # Pattern to match: <<<<<<< HEAD ... ======= ... >>>>>>> backend/djtripx2
    pattern = r'<<<<<<< HEAD\n(.*?)\n=======\n.*?\n>>>>>>> backend/djtripx2'
    resolved = re.sub(pattern, r'\1', content, flags=re.DOTALL)

    conflicts_count = len(re.findall(pattern, content, re.DOTALL))

    if conflicts_count > 0:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(resolved)
        return conflicts_count
    return 0

# Files to process
files = [
    'd:/djtrip/dashbord/src/sections/Page1/Components/LieuDetails.jsx',
    'd:/djtrip/dashbord/src/sections/Page1.jsx',
]

for file_path in files:
    if Path(file_path).exists():
        count = resolve_conflicts_keep_head(file_path)
        if count > 0:
            print(f"✓ {file_path}: Resolved {count} conflicts")
        else:
            print(f"  {file_path}: No conflicts found")
    else:
        print(f"✗ {file_path}: File not found")

print("\nAll conflicts resolved! ✓")
